### PR TITLE
Clean up various GitHub action log messages

### PR DESCRIPTION
**Pull request by bazzadp**
Cleans up a few things in our actions I've noticed recently:
- Removes a couple of warnings on our CodeQL action and runs languages in parallel to hopefully make it faster
- Calibre has moved from `master` to `main` and warns to switch by 1/1/21
- Removes old Generate Chapters action that is no longer required
- Updates linter to latest version as it has some speed benefits (some contributed by yours truly!)
- Renames Pre-Deploy pull request title as no longer just eBooks
- Fixes `pytest` deprecated warning about `getargspec`.

### CODE REVIEW

File: .github/workflows/code-static-analysis.yml [hunk 1]
```diff
@@ -30,48 +28,19 @@ jobs:
   CodeQL-Build:
     # CodeQL runs on ubuntu-latest, windows-latest, and macos-latest
     runs-on: ubuntu-latest
-    # Skip duplicate job for local branches
-    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
+    strategy:
+      matrix:
+        language: ['javascript', 'python']
```
Comment (ibnesayeed) [inline .github/workflows/code-static-analysis.yml:33]: I do not see this matrix variable used anywhere explicitly. Is it something recognized by any steps below implicitly?

Comment (bazzadp) [reply .github/workflows/code-static-analysis.yml:33]: It's used by the CodeQL program though it's not [not very well documented and took me a while to figure out!](https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning#changing-the-languages-that-are-analyzed)

That's what allows this to run in two parallel steps - one for JavaScript and one for Python as can be seen at the bottom of this PR:

image

Normally it runs them together in one step.

Comment (ibnesayeed) [reply .github/workflows/code-static-analysis.yml:33]: I knew how matrix configuration affects parallelism, I was just not aware that this one had an implicit behavior change associated with the specific `language` key. Apparently, the README of the GH repo of the action does not talk about it and I did not check the documentation elsewhere.

**Review by ibnesayeed — APPROVED** [review]
LGTM!

**Pull request merged**